Add docs on custom Operations (#3507)
airbytehq/airbyte: docs/tutorials/transformation-and-normalization/transformations-with-sql.md

# Transformations with SQL \(Part 1/3\)

## Overview

This tutorial will describe how to integrate SQL based transformations with Airbyte syncs using plain SQL queries.

> This is the first part of ELT tutorial. The second part goes deeper with [connecting EL with T using DBT](transformations-with-dbt.md) and then wrap-up with a third part on [connecting EL with T in Airbyte](transformations-with-airbyte.md).

Reviewer: ```suggestion
This is the first part of ELT tutorial. The second part goes deeper with [Transformations with dbt](transformations-with-dbt.md) and then wrap-up with a third part on [Transformations with Airbyte](transformations-with-airbyte.md).
```

This could be useful if:

> 1. You have different usage than analytics that could be handled with these initial data in raw JSON format. 

Reviewer: I might be understanding this one wrong, but I think this clears it up? @ChristopheDuong 
```suggestion
1. You have a use-case not related to analytics that could be handled with data in its raw JSON format. 
```

> 2. You can implement your own Transformer \(even in a different language such as Java or in Spark for example, or another transformation tool: DBT or Dataform\).

Reviewer: ```suggestion
2. You can implement your own transformer. For example, you could write them in a different language, create them in an analytics engine like Spark, or use a transformation tool such as dbt or Dataform.
```

> 1. you have different usage than analytics that could be handled with these initial data in raw JSON format. 

Reviewer 2: ```suggestion
1. You have different usage than analytics that could be handled with these initial data in raw JSON format. 
```

> 2. you can implement your own Transformer \(even in a different language such as Java or in Spark for example, or another transformation tool: DBT or Dataform\) 

Reviewer 2: ```suggestion
2. You can implement your own Transformer \(even in a different language such as Java or in Spark for example, or another transformation tool: DBT or Dataform\).
```

> 3. you want to customize and change how the data is normalized with your own queries

Reviewer 2: ```suggestion
3. You want to customize and change how the data is normalized with your own queries.
```

Airbyte is internally using a specialized tool for handling transformations called DBT.

> An Airbyte Python module reads the `destination_catalog.json` file and  generates DBT code interpreting and transforming the raw data.

Reviewer: ```suggestion
The Airbyte Python module reads the `destination_catalog.json` file and generates dbt code responsible for interpreting and transforming the raw data.
```

> It is made possible thanks to another python-based program that reads the `destination_catalog.json` file and generates code on how to interpret and transform it.

Reviewer 2: ```suggestion
An Airbyte Python module reads the `destination_catalog.json` file and  generates DBT code interpreting and transforming the raw data.
```
